src/main/webapp/script.js

```diff
@@ -797,3 +849,46 @@ function getClosestIndices(indicesList, element) {
   return toReturn;
 }
 
+/**
+ * Sets the value of the mutation slider to the passed amount if it exists
+ * @param {any} amount the desired value of the mutation slider
+ */
+function setMutationSliderValue(amount) {
+  if (!mutationNumSlider) {
+    return;
+  }
+  mutationNumSlider.value = amount;
+}
+
+/**
+ * Updates the maximum, minimum and current values of the graph number
+ * input field
+ */
+function updateGraphNumInput() {
+  const graphNumberInput = document.getElementById("graph-number");
+  graphNumberInput.min = 0;
+  graphNumberInput.max = maxNumMutations;
+  graphNumberInput.value = currMutationNum + 1;
+  const totalMutNumberText = document.getElementById("total-mutation-number-text");
+  if (totalMutNumberText) {
+    totalMutNumberText.innerText = `out of ${maxNumMutations}`;
+  }
+}
+
+/**
+ * Changes the current mutation number based on a change in the value
+ * of the mutation number input
+ */
+function readGraphNumberInput() {
+  const graphNumberInput = document.getElementById("graph-number");
+  if (!graphNumberInput || graphNumberInput.length === 0) {
+    return;
+  }
+  if (graphNumberInput.value >= maxNumMutations) {
+    graphNumberInput.value = maxNumMutations;
+  }
+  if (graphNumberInput.value <= 0) {
+    graphNumberInput.value = 0;
+  }
+  currMutationNum = graphNumberInput.value - 1;
```

Comment:
do you think it could be easier to get rid of this one off? and have the value they put in just be the currMutationNum?

Reply:
I just did it because this input box is part of the "Graph X of XX" display and I thought it would be more intuitive for a user to say "Graph 1" than "Graph 0".